### DISCUSSION TITLE
SofaGym error: "OSError: [Errno 98] Address already in use"

### DISCUSSION THREAD

**Discussion by a8gary** [sofa-framework/sofa#3455] [Write / Run a simulation] [answered]
Hello,

I am trying to run the demo in SofaGym, but I got an error "OSError: [Errno 98] Address already in use", as shown below. It seems the cause is the server tries to reuse the same port to create new clients. Any suggestions? 
![Screenshot from 2022-11-04 12-52-29](https://user-images.githubusercontent.com/106488057/199890596-960631b3-605e-4ca3-90cc-24b031b29bfb.png)
I have Sofa v22.06 on Ubuntu 20.04 and Python3.8. 

Thanks a lot!

**Comment by alxbilger** [comment, marked as answer]
Hi @a8gary,

I think an issue on the SofaGym repository (https://github.com/SofaDefrost/SofaGym/) would be more appropriate.
@EtienneMenager and @p-shg can probably answer your question.

**Reply by EtienneMenager** [reply]
Hi @a8gary, 

It is possible this happens depending on the allocation of addresses and ports during the server / worker connection. Does this block the execution of the program? 

Normally if the address is used already, it should give you an error but find another one to connect and continue the program. 

Let me know :)

**Reply by a8gary** [reply]
Hi @alxbilger, sorry for the wrong place. I will post SofaGym related to SofaGym repo in the future. 

Hi @EtienneMenager, the program stopped there. From `rpc_server.start_server()`, a new available port is assigned only if `port_rpc` is none. 
However, the program can reuse the same port on Win11 OS with no error. But it seems server-to-client communication is messed up.

**Reply by EtienneMenager** [reply]
Thank you for this update. I don't have this problem on ubuntu 20.4 with SOFA 22.06 and python 3.8 ...  I have an address error but it does not stop the program. It is possible that the problem comes from somewhere else (a problem in one of the scenes for example) and that the program stops because of this problem. 

Do you have a complete log of your error?

**Reply by a8gary** [reply]
Thank you. Now I see the program continues after the address error. But I met another error while initiating the Sofa from SofaGym, "libSofaSimulationCommon.so.22.06.00: cannot open shared object file: No such file or directory"
![Screenshot from 2022-11-08 11-15-36](https://user-images.githubusercontent.com/106488057/200466670-57981f44-e0c4-4e98-b997-9d077fea72ca.png)
Sofa works properly if I ran `runSofa` from Sofa installation directory, and all plugins can be loaded with no problem. The Sofa I have is the pre-compiled version downloaded from Sofa release page. Do you have any thoughts? Thanks.

**Reply by a8gary** [reply]
By adding the lib directory to `LD_LIBRARY_PATH` in `.bashrc`, the program is running fine. Thanks for your help.
